wgpu-core/src/device/mod.rs

```diff
@@ -3593,6 +3599,141 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
             .push(id::Valid(render_bundle_id));
     }
 
+    pub fn device_create_query_set<B: GfxBackend>(
+        &self,
+        device_id: id::DeviceId,
+        desc: &wgt::QuerySetDescriptor,
+        id_in: Input<G, id::QuerySetId>,
+    ) -> (id::QuerySetId, Option<resource::CreateQuerySetError>) {
+        span!(_guard, INFO, "Device::create_query_set");
+
+        let hub = B::hub(self);
+        let mut token = Token::root();
+
+        let (device_guard, mut token) = hub.devices.read(&mut token);
+
+        let error = loop {
+            let device = match device_guard.get(device_id) {
+                Ok(device) => device,
+                Err(_) => break DeviceError::Invalid.into(),
+            };
+
+            if matches!(desc.ty, wgt::QueryType::Timestamp)
```

Comment:
please, let's not overuse the fancy stuff. It's clear that you just need one good `match desc.ty` here, not a couple of match macro invocations